### ISSUE TITLE
loading AKAudiofiles into running AKPlayer repeatedly leaks memory

### ISSUE THREAD

**Issue by ghubaddole** [AudioKit/AudioKit#1826]
simple setup - AKPlayer -> output.
have an AKAudioFile object, initialized from a file on disk.

in a loop, I stop the player, load the file, prepare() and then play(). watch as memory usage grows fast.

2 similar issues were fixed in the past: 1020 and 1792.

**Comment by ghubaddole** [comment]
<img width="1151" alt="Screen Shot 2019-08-08 at 11 36 27 AM" src="https://user-images.githubusercontent.com/46110201/62679100-0f72dc80-b9d1-11e9-8ca3-c3ba9b7d36c8.png">

 memory graph. AVAudioChannelLayout.layout (101678).

**Comment by ghubaddole** [comment]
2 Leaks : one in AKAudioFile, one in AKPlayer

I tried the following:
`
var file : AKAudioFile? = nil
for _ in 0...1000000 {
   file =  try AKAudioFile(readFileName: "some/path/to/wav") // <<<--- leaks NSURL
} 
 `
this code is leaking a NSURLs for each iteration.

then I tried:

`
let file : AKAudioFile  =  try AKAudioFile(readFileName: "some/path/to/wav")
let player : AKPlayer() // connected to output
for _ in 0...1000000 {
    player.load(audioFile: file)  // this leaks AVAudioChannelLayout.layout.
} 

As a workaround, I tried AKWaveTable, which works fine with no leaks, but crashes if engine restarts due to a route/config change.

**Comment by ryanfrancesconi** [comment]
I'll look into this asap . thank you!

**Comment by ryanfrancesconi** [comment]
As a work around, I recommend instantiating a new AKPlayer when the file changes. Also, use the url: constructor for AKPlayer. Something like this:

`    /// open an audio URL for playing
    func open(url: URL) {
        // for now just make a new player so it's not necessary to rebalance format types
        // if the processingFormat changes
        if player != nil {
            player?.detach()
            player = nil
        }

        AKLog("Creating player...", url)
        player = AKPlayer(url: url)

    }`

**Comment by aure** [comment]
I think @ryanfrancesconi 's answers effectively close this issue.
